lib/Dialect/FIRRTL/FIRRTLFolds.cpp

```diff
@@ -3194,6 +3194,24 @@ static LogicalResult eraseIfZeroOrNotZero(Operation *op, Value predicate,
   return failure();
 }
 
+namespace {
+/// Erase a verif op if the predicate and enable are the same.
+template <typename T>
+struct PredicateEqToEnable : mlir::OpRewritePattern<T> {
+  using OpRewritePattern<T>::OpRewritePattern;
+
+  LogicalResult matchAndRewrite(T op,
+                                PatternRewriter &rewriter) const override {
+    if (op.getPredicate() == op.getEnable()) {
+      rewriter.eraseOp(op);
+      return success();
+    }
+
+    return failure();
+  }
+};
```

Comment:
Challenge 1: Can this be written using ODS or does that not allow op removal?

Reply:
I honestly didn't think it was possible, but turned out to be quite easy 👍